aws/aws.go

```diff
@@ -30,6 +31,13 @@ type Region struct {
 	Sign                 Signer // Method which will be used to sign requests.
 }
 
+func (r Region) ResolveS3BucketEndpoint(bucketName string) string {
+	if r.S3BucketEndpoint != "" {
+		return strings.Replace(r.S3BucketEndpoint, "${bucket}", bucketName, -1)
+	}
+	return r.S3Endpoint + "/" + bucketName + "/"
```

Comment:
path.Join scrubs the content and will remove one of the slashes in "http://" in the r.S3Endpoint.